internal/api/mfa.go

```diff
@@ -399,17 +388,49 @@ func (a *API) ChallengeFactor(w http.ResponseWriter, r *http.Request) error {
 	})
 }
 
-func (a *API) verifyPhoneFactor(w http.ResponseWriter, r *http.Request, params *VerifyFactorParams) error {
+func (a *API) ChallengeFactor(w http.ResponseWriter, r *http.Request) error {
 	ctx := r.Context()
 	config := a.config
+	factor := getFactor(ctx)
+	user := getUser(ctx)
+
+	if !factor.IsOwnedBy(user) {
```

Comment:
This is check is a change which is not a refactor. Added this check to prevent users from sending SMS-es to unknown numbers.

Reply:
should this check only be done after the `MFA.Phone.VerifyEnabled` check below?

Reply:
While it's not strictly needed and unlikely to occur would like to block TOTP factors from calling Challenge on factors they don't own as well.

Can move if there's a strong preference

Reply:
Yeah I'd move it up, because if the flow is disabled the first thing to check it should be if it's disabled. Shouldn't leak information about what is owned by whom.

Reply:
Thanks! Expanding on Kang's suggestion maybe we can also move the enabled / disabled check into `loadFactor`. I will move the check down for this PR and then follow up with a refactor right after.